### PR TITLE
Refs [Task][30556] Remote data source

### PR DESCRIPTION
**Pull request by NguyenTaTrung**
## Related Tickets
- [#Task_30556: Remote data source](https://edu-redmine.sun-asterisk.vn/issues/30556)

## WHAT
- Create remote data source

## HOW
- Create data class
- Create request ready to get data
- Create pakage, class apply repository pattern

## Review Checklist

Category | View Point | Description | Expected Reviewer Answer | Self review | Reviewer2 (name)
--- | --- | --- | --- | --- | ---
Conventions | Does the code follow Sun* coding style and coding conventions? | https://github.com/framgia/coding-standards/tree/master/eng/android | YES |<li>- [x] yes</li>|<li>- [ ] yes</li>  
Redmine | Does the ticket follow Sun* Redmine working process?  | https://github.com/framgia/Training-Guideline/blob/master/WorkingProcess/redmine/redmine.md| YES |<li>- [x] yes</li>|<li>- [ ] yes</li>  
Documentation | Is there any incomplete code? If so, should it be removed or flagged with a suitable marker like ‘TODO’? |  | YES |<li>- [x] yes</li>|<li>- [ ] yes</li>  
## Notes (Optional)
*(Impacted Areas in Application(List features, api, models or services that this PR will affect))*

*(List gem, library third party add new)*

*(Other notes)*

### CODE REVIEW

File: app/src/main/java/com/example/streetworkout/data/model/GroupMuscle.kt [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    @SerializedName(IMAGE)
+    val image: String,
+    @SerializedName(LEVEL)
+    val level: Int
+) {
+    companion object {
+        private const val TITLE = "title"
+        private const val IMAGE = "image"
+        private const val LEVEL = "level"
+        private const val ID = "id"
```
Comment (huytq-2422) [inline app/src/main/java/com/example/streetworkout/data/model/GroupMuscle.kt]: không cần phải constant đâu

File: app/src/main/java/com/example/streetworkout/data/repository/CategoryRepository.kt [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+) : CategoryDataSource.Remote {
+
+    override fun getInformationHome(): Observable<Category> = remote.getInformationHome()
+
+    companion object {
+        private var instance: CategoryRepository? = null
+        fun getInstance(local: CategoryDataSource.Local, remote: CategoryDataSource.Remote) =
+            instance ?: CategoryRepository(local, remote).also { instance = it }
+    }
+}
```
Comment (huytq-2422) [inline app/src/main/java/com/example/streetworkout/data/repository/CategoryRepository.kt]: Ông dùng Koin để dependence inject đi ôn g

File: app/src/main/java/com/example/streetworkout/data/resource/remote/util/RetrofitClient.kt [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+    companion object {
+        private var instance: Retrofit? = null
+        fun getInstance(url: String): Retrofit =
+            instance ?: Retrofit.Builder()
+                .baseUrl(url)
+                .addCallAdapterFactory(RxJava3CallAdapterFactory.create())
+                .addConverterFactory(GsonConverterFactory.create())
+                .build().also { instance = it }
+    }
+}
```
Comment (huytq-2422) [inline app/src/main/java/com/example/streetworkout/data/resource/remote/util/RetrofitClient.kt]: Sao lại truyền url vào nhỉ, dùng nhiều api à ông
```suggestion
object ApiFactory {
        fun getInstance(url: String): Retrofit =
            instance ?: Retrofit.Builder()
                .baseUrl(url)
                .addCallAdapterFactory(RxJava3CallAdapterFactory.create())
                .addConverterFactory(GsonConverterFactory.create())
                .build()
    }
}
```

**Review by datdt-1935 — APPROVED** [review]

File: app/src/main/java/com/example/streetworkout/data/repository/CategoryRepository.kt (hunk 2 above)
Comment (NguyenTaTrung) [reply app/src/main/java/com/example/streetworkout/data/repository/CategoryRepository.kt]: Ok ông

File: app/src/main/java/com/example/streetworkout/data/resource/remote/util/RetrofitClient.kt (hunk 3 above)
Comment (NguyenTaTrung) [reply app/src/main/java/com/example/streetworkout/data/resource/remote/util/RetrofitClient.kt]: Có 1 api về food nữa t đang suy nghĩ xem có lấy, t định tự viết vì api kia k ổn cho lắm

**Pull request merged**